enacts/calc.py

```diff
@@ -619,6 +619,42 @@ def daily_tobegroupedby_season(
     return daily_tobegroupedby_season
 
 
+def seasonal_groups(
+    time_coord, start_day, start_month, end_day, end_month
+):
+    """Create seasonal groups identified by first day of each season"""
+    if start_day == 29 and start_month == 2:
+        start_day = 1
+        start_month = 3
+    start_edges = sel_day_and_month(time_coord, start_day, start_month)
+    if end_day == 29 and end_month == 2:
+        end_edges = sel_day_and_month(time_coord, 1 , 3, offset=-1)
+    else:
+        end_edges = sel_day_and_month(time_coord, end_day, end_month)
+    if end_edges[0] < start_edges[0]:
+        end_edges = end_edges[1:]
+    if start_edges[-1] > end_edges[-1]:
+        start_edges = start_edges[:-2]
```

Comment:
I pushed a proposed solution to this. Have a look.

Reply:
I think you've fixed the problem. Not sure it's the simplest solution--what if we just generated a dense T coordinate, created start and end edges based on that, and then selected from the dense coordinate using the indices of the sparse one? Implementation detail and thus of secondary concern.

Reply:
just pushed something else.

Reply:
The implementation still seems unnecessarily complicated, but that's something to work together on interactively rather than via PR comments, and doesn't need to block progress. What's most important is the interface and the tests.

Regarding the interface, either assign_season_coords should be renamed select_season (select_and_label_season?), or it should only label seasons and not drop data outside of the season (set season start and end to NaT outside the season).

Regarding tests, here are some properties of the function that could be tested:
* Behavior when start of the data is before the season, during the season, after the season. Likewise at the end of the data.
* Confirm that all the seasons in the range of the data have been labeled/retained. If we mistakenly considered all of the data points to be out of season, the assertion you're currently using would pass.
* Confirm that season ends are all the right number of days after season starts, i.e. that we haven't accidentally created seasons that span multiple years.

Reply:
> Behavior when start of the data is before the season, during the season, after the season. Likewise at the end of the data.

Duh, "before the season" and "after the season" are the same thing, better called "outside the season."

Which reminds me, we really need a test (or tests) for a season that spans Jan 1. That case has been a very frequent source of errors in our various python applications.

Reply:
I think all that was mentioned here has been addressed? except:

"Confirm that all the seasons in the range of the data have been labeled/retained. If we mistakenly considered all of the data points to be out of season, the assertion you're currently using would pass."

You're saying that all my tests are confirming that what I retained is within their appropriate seasons, but I am not testing that I did retain all that I should have?

Reply:
Yes, that seems to be what I was saying.

Reply:
I can't wrap my head around what the test(s) could be...

Reply:
Construct the expected T coordinate, assert that the T coordinate returned by the function is the same?

Reply:
I pushed something to that effect